File: app/FizzBuzz/FizzBuzzCommand.cs
namespace FizzBuzz;

public interface IFizzBuzzCommand
{
    FizzBuzzValue Execute(Number number);
}

public class FizzBuzzCommand : IFizzBuzzCommand
{
    private readonly FizzBuzzType _type;

    public FizzBuzzCommand(FizzBuzzType type)
    {
        _type = type;
    }

    public FizzBuzzValue Execute(Number number)
    {
        return _type.Generate(number);
    }
}

File: app/lib/application/fizz_buzz_command.py
"""FizzBuzz command interface module."""

from abc import ABC, abstractmethod


class FizzBuzzCommand(ABC):
    """Abstract base class for FizzBuzz commands."""

    @abstractmethod
    def execute(self, number: int) -> object:
        """Execute the command.

        Args:
            number: The number to process

        Returns:
            Result of the command execution
        """
        pass
